crates/router/src/services/api.rs

```diff
@@ -312,7 +313,13 @@ async fn send_request(
                 // Currently this is not used remove this if not required
                 // If using this then handle the serde_part
                 Some(ContentType::FormUrlEncoded) => {
-                    let url_encoded_payload = serde_urlencoded::to_string(&request.payload)
+                    let payload = match request.payload.clone() {
+                        Some(req) => serde_json::from_str(req.peek())
```

Comment:
is this conversion is really necessary? What happens if we directly try `serde_urlencoded::to_string(req)` incase of some and serde_urlencoded:: to_string(Secret::new("")) incase of None? 

If we can directly convert the Secret<String> to url_encoded string we can skip the `peek` and `serde_json::from_str` part

Reply:
```2023-04-05T10:53:49.131754Z ERROR router::services::api: error: {"error":{"type":"api","message":"Something went wrong","code":"HE_00"}}
├╴at crates/router/src/services/api.rs:517:42
│
├─▶ {"error":{"type":"server_not_available","code":"HE_00","message":"Something went wrong"}}
│   ╰╴at crates/router/src/core/payments/flows/authorize_flow.rs:102:44
│
├─▶ Failed to execute a processing step: None
│   ╰╴at crates/router/src/services/api.rs:266:30
│
├─▶ URL encoding of request payload failed
│   ├╴at crates/router/src/services/api.rs:317:26
│   ╰╴Unable to do url encoding on request: Some(*** alloc::string::String ***)
│
╰─▶ top-level serializer supports only maps and structs```
You will get above error saying maps and struct only can be urlencoded. That's why we are converting it to JSON first and then url_encode